src/ui/hotkey_overlay.rs

```diff
@@ -285,6 +285,21 @@ fn render(
         }
     }
 
+    if config.hotkey_overlay.hide_not_bound {
+        // If hiding unbound hotkeys is desired
+        // filter out actions that aren't present in the binds
+        actions = actions
+            .into_iter()
+            .filter(|action| {
+                binds
+                    .iter()
+                    .map(|bind| bind.action.clone())
+                    .collect::<Vec<Action>>()
+                    .contains(action)
```

Comment:
This is collecting all binds into a freshly allocated Vector for every action that you check. What you need is something along `.iter().any(|bind| bind.action == action)`

Reply:
ah, yes of course.

i was thinking perhaps it'd be nice to do this filtering on 
```rust
let strings = actions
        .into_iter()
        // .. something clever here
        .filter_map(|action| format_bind(binds, mod_key, action))
        .collect::<Vec<_>>();
```

but perhaps that's not necessary. :)